From 4fc77e4f2cd1740aef5cec092e7df7af6a6fbeb4 Mon Sep 17 00:00:00 2001
From: Marc Zyngier <marc.zyngier@arm.com>
Date: Thu, 8 Sep 2011 13:15:22 +0100
Subject: [PATCH 0569/1083] ARM: SoC: convert shmobile SMP to SMP operations

Convert shmobile SMP platforms to use struct smp_operations to provide
their SMP and CPU hotplug operations.

Cc: Paul Mundt <lethal@linux-sh.org>
Cc: Magnus Damm <magnus.damm@gmail.com>
Signed-off-by: Marc Zyngier <marc.zyngier@arm.com>
Acked-by: Nicolas Pitre <nico@linaro.org>
Signed-off-by: Arnd Bergmann <arnd@arndb.de>

Conflicts:
	arch/arm/mach-shmobile/platsmp.c
	arch/arm/mach-shmobile/smp-sh73a0.c

[ horms+renesas@verge.net.au: This patch appears a second time
  in the backport to supply a portion that was not applicable
  earlier as "ARM: GIC: remove direct use of gic_raise_softirq"
  had not been applied. ]

Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit dc333259c7c3a33550fabbdf8597d319512791e8)

Signed-off-by: Ryo Kataoka <ryo.kataoka.wt@renesas.com>
---
 arch/arm/mach-shmobile/platsmp.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/arch/arm/mach-shmobile/platsmp.c b/arch/arm/mach-shmobile/platsmp.c
index 4d1142f..5833c60 100644
--- a/arch/arm/mach-shmobile/platsmp.c
+++ b/arch/arm/mach-shmobile/platsmp.c
@@ -31,8 +31,3 @@ void __init shmobile_smp_init_cpus(unsigned int ncores)
 	for (i = 0; i < ncores; i++)
 		set_cpu_possible(i, true);
 }
-
-void __init platform_smp_prepare_cpus(unsigned int max_cpus)
-{
-	shmobile_smp_prepare_cpus();
-}
-- 
1.8.3.2

